symbulate/tests/test_distributions.py

```diff
@@ -10,25 +10,705 @@
-        X = RV(Bernoulli(.4) ** 5)
+        exp_list, obs_list = [], []
+        X = RV(Bernoulli(p=.4) ** 5)
         sims = X.apply(sum).sim(Nsim)
-        obs = list(sims.tabulate().values())
-        exp = Nsim * stats.binom(n=5, p=.4).pmf(range(6))
-        pval = stats.chisquare(obs, exp).pvalue
+        key_list = sims.tabulate()
+        for k in range(6):
+            expected = Nsim * stats.binom(n=5, p=.4).pmf(k)
+            if expected > 5:
+                exp_list.append(expected)
+                obs_list.append(key_list[k])
+        pval = stats.chisquare(obs_list, exp_list).pvalue
+        self.assertTrue(pval > 0.01)
+
+    def test_Bernoulli_Binomial_n_1(self):
+        exp_list, obs_list = [], []   
+        X = RV(Bernoulli(p=.4))
+        sims = X.sim(Nsim)
+        key_list = sims.tabulate()
+        for k in range(2): 
+            expected = Nsim * stats.binom(n=1, p=.4).pmf(k)
+            if expected > 5:
+                exp_list.append(expected)
+                obs_list.append(key_list[k])
+        pval = stats.chisquare(obs_list, exp_list).pvalue
+        self.assertTrue(pval > 0.01)
+
+        
+class TestBinomial(unittest.TestCase):
+    
+    def test_Binomial_p_1(self):
+        for nsample in range(1, 1000, 100):
+            X = RV(Binomial(n=nsample, p=1.0))
+            sims = X.sim(Nsim)
+        self.assertTrue(all(sim == nsample for sim in sims))
+
+    def test_Binomial_error_n(self):
+        self.assertRaises(Exception, lambda: Binomial(n=-10, p=0.4))
+    
+    def test_Binomial_additive(self):
+        exp_list, obs_list = [], []
+        X, Y = RV(Binomial(n=8, p=0.6) * Binomial(n=5, p=0.6))
+        sims = (X & Y).sim(Nsim).apply(sum)
+        key_list = sims.tabulate()
+        for k in range(2, 14):
+            expected = Nsim * stats.binom(n=13, p=.6).pmf(k)
+            if expected > 5:
+                exp_list.append(expected)
+                obs_list.append(key_list[k])
+        pval = stats.chisquare(obs_list, exp_list).pvalue   
+        self.assertTrue(pval > 0.01)
+    
+
+class TestHypergeometric(unittest.TestCase):
+
+    def test_Hypergeometric_no_failures(self):
+        X = RV(Hypergeometric(n=10, N0=0, N1=1000))
+        sims = X.sim(Nsim)
+        self.assertTrue(all(sim == 10 for sim in sims))
+
+    def test_Hypergeometric_Binomial_converge(self):
+        exp_list, obs_list = [], []
+        X = RV(Hypergeometric(n=8, N0=200, N1=800))
+        sims = X.sim(Nsim)
+        key_list = sims.tabulate()
+        for k in range(9):
+            expected = Nsim * stats.binom(n=8, p=.8).pmf(k)
+            if expected > 5:
+                exp_list.append(expected)
+                obs_list.append(key_list[k])
+        pval = stats.chisquare(obs_list, exp_list).pvalue
+        self.assertTrue(pval > 0.01)
+
+    def test_Hypergeometric_error(self):
+        self.assertRaises(Exception, lambda: 
+                                     Hypergeometric(n=10, N0=0, N1=8))
```

Comment:
Is it raising an error because N0 is 0 (it shouldn't) or because (N0 + N1) < n? This is not a good example because there are potentially many reasons that this is throwing an error.